common/cpp/src/google_smart_card_common/value_debug_dumping.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_SMART_CARD_COMMON_VALUE_DEBUG_DUMPING_H_
+#define GOOGLE_SMART_CARD_COMMON_VALUE_DEBUG_DUMPING_H_
+
+#include <string>
+
+#include <google_smart_card_common/value.h>
+
+namespace google_smart_card {
+
+// Generates a sanitized debug representation of the given value. In Release
+// builds, this only dumps the title of the value's type; in Debug builds, this
+// is equivalent to `DebugDumpValueFull()`.
+std::string DebugDumpValueSanitized(const Value& value);
+
+// Generates a full debug representation of the given value.
+//
+// NOTE: It's dangerous to use this function with variables that might
+// potentially contain privacy/security sensitive data. Use
+// `DebugDumpValueSanitized()` instead.
+std::string DebugDumpValueFull(const Value& value);
```

Comment:
Would it make sense to only define this if NDEBUG is not set?

Reply:
Ack - The idea is that this function should be used in places where the caller knows for sure that the data is non-sensitive.

For example, the existing code does this when logging parameters for the `chrome.certificateProvider.requestPin()` method: https://github.com/GoogleChromeLabs/chromeos_smart_card_connector/blob/4d02c5320eb824fad87e377d28e2fb87218180aa/example_cpp_smart_card_client_app/src/chrome_certificate_provider/api_bridge.cc#L140
(A side note: In the existing code these `pp::Var`-based helpers are named `DumpVar()` and `DebugDumpVar()`, but I guess these were not really informative names. Hopefully, the naming proposed in this PR will be less ambiguous and harder to confuse.)